core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -475,6 +475,19 @@ object SparkSubmit extends CommandLineUtils with Logging {
       }
     }
 
+    // for pyspark virtualenv
+    if (args.isPython) {
+      if (clusterManager != YARN &&
+        args.sparkProperties.getOrElse("spark.pyspark.virtualenv.enabled", "false") == "true") {
+        printErrorAndExit("virtualenv is only supported in yarn mode")
```

Comment:
Why doesn't this work in standalone?

Reply:
I haven't tested it in standalone mode, so not guaranteed for that, it is on my plan to support it for standalone later.

Reply:
Ah please make a JIRA to track this then.